lib/Jsrt/JsrtDebugUtils.cpp

```diff
@@ -389,6 +394,16 @@ void JsrtDebugUtils::AddPropertyToObject(Js::DynamicObject * object, JsrtDebugPr
     JsrtDebugUtils::AddVarPropertyToObject(object, propertyId, value, scriptContext);
 }
 
+bool JsrtDebugUtils::HasProperty(Js::DynamicObject * object, JsrtDebugPropertyId propertyId, Js::ScriptContext * scriptContext)
+{
+    const char16* propertyName = GetDebugPropertyName(propertyId);
+
+    const Js::PropertyRecord* propertyRecord;
+    scriptContext->GetOrAddPropertyRecord(propertyName, static_cast<int>(wcslen(propertyName)), &propertyRecord);
+
+    return !!Js::JavascriptOperators::HasProperty(object, propertyRecord->GetPropertyId());
```

Comment:
Also, you probably don't need GetOrAddPropertyRecord here- if FindPropertyRecord returns null, then you could return false here. There's no need to do another allocation here.

---
In reply to: [111523435](https://github.com/Microsoft/ChakraCore/pull/2832#discussion_r111523435) [](ancestors = 111523435)